### ISSUE TITLE
Installation issue: openjdk

### ISSUE THREAD

**Issue by serikkehva** [LLNL/spack#35171]
### Steps to reproduce the issue

```console
$ spack spec -I openjdk
$ spack install openjdk
```

### Error message

<details><summary>Error message</summary><pre>
==> Installing openjdk-11.0.17_8-ea25qbpom5opt65u3m7cxlpzo746az6c
==> No binary for openjdk-11.0.17_8-ea25qbpom5opt65u3m7cxlpzo746az6c found: installing from source
==> Using cached archive: /mnt/storage_2/scratch/grant_531/spack/var/spack/cache/_source-cache/archive/b8/b8d46ed08ef4859476fe6421a7690d899ed83dce63f13fd894f994043177ef3c.tar.gz
==> No patches needed for openjdk
==> openjdk: Executing phase: 'install'
==> Error: PermissionError: [Errno 13] Permission denied: '/mnt/storage_2/scratch/grant_531/spack/opt/spack/linux-centos7-cascadelake/gcc-10.2.0/openjdk-11.0.17_8-ea25qbpom5opt65u3m7cxlpzo746az6c/legal/jdk.dynalink/dynalink.md'

/mnt/storage_2/scratch/grant_531/spack/var/spack/repos/builtin/packages/openjdk/package.py:382, in install:
        380    def install(self, spec, prefix):
        381        top_dir = "Contents/Home/" if platform.system() == "Darwin" else "."
  >>    382        install_tree(top_dir, prefix)

</pre></details>

### Information on your system

* **Spack:** 0.19.0
* **Python:** 3.6.8
* **Platform:** linux-centos7-cascadelake
* **Concretizer:** clingo

### Additional information

==> openjdk: Executing phase: 'install'
==> [2023-01-26-08:27:10.932695] Installing . to /mnt/storage_2/scratch/grant_531/spack/opt/spack/linux-centos7-cascadelake/gcc-10.2.0/openjdk-11.0.17_8-ea25qbpom5opt65u3m7cxlpzo746az6c

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [X] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

I have this problem installing spack on HPC system, on which I'm only a user. Locally on my computer, the installation goes fine. I've seen a similar problem reported on the internet but without a solution.

**Comment by mwkrentel** [comment]
@serikkehva What system?
What type of file system is this on?

This sounds like the "python 3.6 on lustre" bug.
#25354 and #27247 which is not yet merged.

**Comment by serikkehva** [comment]
Yes indeed, thank you for pointing me in the right direction. Changing SPACK_PYTHON to python3.9.2 solved my problem!
